app/scripts/app.js

```diff
@@ -55,10 +95,16 @@ app.controller('ModalInstanceCtrl', ['$scope', '$modalInstance', 'Room', functio
         rooms.$add({ displayName: roomName }).then(function(firebaseRef) {
         var id = firebaseRef.key();
         console.log("Added record with id " + id);
+        // seems like you'd ideally want a prompt here to capture the username
```

Comment:
I think you may be over thinking this.  Think about possibly creating a singleton (similar to Room) that contains user information.  I'm not sure you want to force a user to set their username every time they enter a room.  Consider creating a button to allow a user to change their username that is outside the context of any particular room.  Leverage the singleton to contain the username/user information (first name, last name, etc), and then each room can depend on that singleton, giving them access to the username.  As the user changes their username, each room will automagically be updated because they are referencing the singleton.